crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -475,6 +475,7 @@ impl<F: Send + Clone, Ctx: PaymentMethodRetrieve>
         let m_merchant_account = merchant_account.clone();
         let m_request = request.clone();
         let m_key_store = key_store.clone();
+        let m_payment_method_id = payment_attempt.payment_method_id.clone();
```

Comment:
`payment_method_id` can also be passed in the confirm request right? why are we always inferring it from `payment_attempt`.

Reply:
Yes, right we won't need to infer it in the update and confirm. Will  pass None